lib/services/provided/autocomplete.js

```diff
@@ -39,29 +41,41 @@ function parseCompletions(results: CompleteReply, prefix: string) {
 
   if (metadata && metadata._jupyter_types_experimental) {
     const comps = metadata._jupyter_types_experimental;
-    if (
-      comps.length > 0 &&
-      comps[0].text != null &&
-      comps[0].start != null &&
-      comps[0].end != null
-    ) {
-      return _.map(comps, match => ({
-        text: match.text,
-        replacementPrefix: prefix.slice(match.start, match.end),
-        type: match.type,
-        iconHTML:
-          !match.type || match.type === "<unknown>" ? iconHTML : undefined
-      }));
+    if (comps.length > 0 && comps[0].text) {
+      return _.map(comps, match => {
+        const text = match.text;
```

Comment:
after this can we add
```
const start = match.start ? match.start : cursor_start
const end = match.end ? match.end : cursor_end
```
then update all the use cases below it makes it more readable

Reply:
Done in 0b616ed

Reply:
No lol. Thats not what I meant can you revert.

Reply:
Umm I doubt that we can have an assumption that `match.start` always comes with `match.end`, thus this kinda condition check would be preferred here.

Reply:
```javascript
        const start = match.start && match.end ? match.start : cursor_start
        const end= match.start && match.end ? match.end: cursor_end
        const replacementPrefix = prefix.slice(start, end);
        const replacedText = prefix.slice(0, start) + text;
```

Reply:
You may miss the case like below. I suspect it never happens, but if it does, it would cause some very annoying error.
```js
start = match.start
end = cursor_end
```

Reply:
Look above @aviatesk

Reply:
Much neater finally :)